tests/testapp/tests/test_syncsession_methods.py

```diff
@@ -0,0 +1,251 @@
+
+    def test_partition_and_fsic_buffering(self):
+        self.filter_prefixes = ['{}:user:summary'.format(self.data['user1'].id)]
+        self.fsics = {self.data['group1_id']: 1}
+        self.data['sc']._queue_into_buffer(self.filter_prefixes, self.fsics)
+        buffer_ids = Buffer.objects.values_list('model_uuid', flat=True)
+        rmcb_ids = RecordMaxCounterBuffer.objects.values_list('model_uuid', flat=True)
+        for i in self.data['user1_sumlogs']:
+            self.assertIn(i.id, buffer_ids)
+            self.assertIn(i.id, rmcb_ids)
+        for i in self.data['user2_sumlogs']:
+            self.assertNotIn(i.id, buffer_ids)
+            self.assertNotIn(i.id, rmcb_ids)
+        for i in self.data['user3_sumlogs']:
+            self.assertNotIn(i.id, buffer_ids)
+            self.assertNotIn(i.id, rmcb_ids)
+
+
+class BufferIntoStoreTestCase(TestCase):
+
+    def setUp(self):
+        self.data = {}
+        DatabaseIDModel.objects.create()
+        (self.current_id, _) = InstanceIDModel.get_or_create_current_instance()
+
+        # create controllers for app/store/buffer operations
+        self.data['mc'] = MorangoProfileController('facilitydata')
+        self.data['sc'] = SyncClient('host', 'facilitydata')
+
+        self.data.update(create_buffer_and_store_dummy_data(self.data['sc'].transfer_session_id))
+
+    def test_dequeuing_delete_rmcb_records(self):
+        for i in self.data['model1_rmcb_ids']:
+            self.assertTrue(RecordMaxCounterBuffer.objects.filter(instance_id=i, model_uuid=self.data['model1']).exists())
+        with connection.cursor() as cursor:
+            self.data['sc']._dequeuing_delete_rmcb_records(cursor)
+        for i in self.data['model1_rmcb_ids']:
+            self.assertFalse(RecordMaxCounterBuffer.objects.filter(instance_id=i, model_uuid=self.data['model1']).exists())
+
+    def test_dequeuing_delete_buffered_records(self):
+        self.assertTrue(Buffer.objects.filter(model_uuid=self.data['model1']).exists())
+        with connection.cursor() as cursor:
+            self.data['sc']._dequeuing_delete_buffered_records(cursor)
+        self.assertFalse(Buffer.objects.filter(model_uuid=self.data['model1']).exists())
+
+    def test_dequeuing_merge_conflict_rmc(self):
+        rmc = RecordMaxCounter.objects.get(instance_id=self.data['model2_rmc_ids'][0], store_model_id=self.data['model2'])
+        rmcb = RecordMaxCounterBuffer.objects.get(instance_id=self.data['model2_rmc_ids'][0], model_uuid=self.data['model2'])
+        self.assertNotEqual(rmc.counter, rmcb.counter)
+        with connection.cursor() as cursor:
+            self.data['sc']._dequeuing_merge_conflict_rmc(cursor)
+        rmc = RecordMaxCounter.objects.get(instance_id=self.data['model2_rmc_ids'][0], store_model_id=self.data['model2'])
+        rmcb = RecordMaxCounterBuffer.objects.get(instance_id=self.data['model2_rmc_ids'][0], model_uuid=self.data['model2'])
+        self.assertEqual(rmc.counter, rmcb.counter)
```

Comment:
Seems like there are three cases that would be good to test; not certain whether all are captured here:
- rmcb.instance_id isn't in rmc
- rmcb.counter is less than or equal to rmc.counter
- rmcb.counter is greater than rmc.counter